react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { password: '' };
+    this.handleInput = this.handleInput.bind(this);
+    this.getErrorMessage = this.getErrorMessage.bind(this);
+  }
+
+  handleInput(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  getErrorMessage() {
+    if (this.state.password.length <= 7) {
+      return 'password is too short';
+    } else {
+      return '';
+    }
```

Comment:
Try moving your "required" error message into here too so you don't have duplicated react elements.